### ISSUE TITLE
does Taskcat support testing existing stacks?

### ISSUE THREAD

**Issue by kidbrax** [aws-quickstart/taskcat#638]
Not a bug, more of a support question. Can Taskcat test existing stacks? I'd like to be able to create my own stacks and just test those resources instead of taskcat deploying the cfn and then testing. It's unclear from the docs if that is possible.

**Comment by andrew-glenn** [comment]
Perhaps. Would you mind expanding on your use case? 

There's a roadmap item I'm mulling around that would let you supply an S3 URL for a template, then pass that into CFN with parameters you specify. The idea is that taskcat could act as a canary of sorts, but I'm not certain that's what you're looking for.

**Comment by kidbrax** [comment]
I'm looking for something similar to what Inspec for AWS provides. I give it a list of resources(and how they're configured) and it queries AWS for the resources to see if they exist and match the config required. So I deploy my stack and then run my tests and they should pass. Does that help?

**Comment by andrew-glenn** [comment]
Yeah. You'll probably benefit from pre/post hooks. The (current) value-add of taskcat is to test the CFN stack itself. 

The pre/post-hook feature would enable you to invoke an external process post-SUCCESS to run any additional tests you may wish to run. (Verifying instance X can reach API Y, etc)

Make sense?

**Comment by kidbrax** [comment]
I'm not sure if we're talking about the same thing. Let's say I just deploy a resource by hand, could Taskcat verify it after I deploy the resource?

**Comment by andrew-glenn** [comment]
No. Taskcat is a CloudFormation testing framework. It has no knowledge of resources created via other means (terraform, by hand, etc).
